connectors/ssh/src/sessions/session.cc

```diff
@@ -672,17 +673,17 @@ void session::handshake_handler(int retval,
 
     // Check fingerprint.
     if (check != LIBSSH2_KNOWNHOST_CHECK_MATCH) {
-      exceptions::basic e(basic_error());
-      e << "host '" << _creds.get_host()
-        << "' is not known or could not be validated: ";
+      std::string e;  // exception complement
       if (LIBSSH2_KNOWNHOST_CHECK_NOTFOUND == check)
-        e << "host was not found in known_hosts file " << known_hosts_file;
+        e = fmt::format("host was not found in known_hosts file {}",
+                        known_hosts_file);
       else if (LIBSSH2_KNOWNHOST_CHECK_MISMATCH == check)
-        e << "host fingerprint mismatch with known_hosts file "
-          << known_hosts_file;
+        e = fmt::format("host fingerprint mismatch with known_hosts file {}",
+                        known_hosts_file);
       else
-        e << "unknown error";
-      throw e;
+        e = "unknown error";
+      throw msg_fmt("host '{}' is not known or could not be validated: {}",
+                    _creds.get_host(), e);
```

Comment:
In the three cases, we throw an exception.
I propose to throw it directly in the three cases, so no need of e

Reply:
will fix